tmp/optimization.py

```diff
@@ -261,6 +257,76 @@ def _start_optimization(self, npoints):
         print('Best result is: ' + str(best))
         print('Figure of merit is: ' + str(np.min(f)))
 
+
+class GA():
+
+    def __init__(self, npop, nparents):
+        self._lower_limits = np.array([])
+        self._upper_limits = np.array([])
+        self._indiv = np.array([])
+        self.initialization()
+        self._ndim = len(self._upper_limits)
+        self._npop = npop
+        self._nparents = nparents
+        self._nchildren = self._npop - self._nparents
+        self._crosspoint = int(self._ndim/2)
+        pass
+
+    def initialization(self):
+        pass
+
+    def _check_initialization(self):
+        pass
+
+    def calc_merit_function(self):
+        return 0
+
+    def _create_pop(self):
+        dlim = self._upper_limits - self._lower_limits
+        rarray = np.random.rand(self._npop, self._ndim)
+        self._indiv = dlim * rarray + self._lower_limits
+
+    def _select_parents(self, f):
+        ind_sort = np.argsort(f)
+        return self._indiv[ind_sort[:self._nparents], :]
+
+    def _crossover(self, parents):
+        child = np.zeros([self._nchildren, self._ndim])
+        for i in range(self._nchildren):
+            dad = i % self._nparents
+            mom = (i+1) % self._nparents
+            child[i, :self._crosspoint] = parents[dad, :self._crosspoint]
+            child[i, self._crosspoint:] = parents[mom, self._crosspoint:]
+        return child
+
+    def _mutation(self):
```

Comment:
I think mutation should be implemented even in this first version of the algorithm.